src/positions/saga.ts

```diff
@@ -188,6 +201,65 @@ export function* handleEnableHooksPreviewDeepLink(
   }
 }
 
+export function* triggerShortcutSaga({ payload }: ReturnType<typeof triggerShortcut>) {
+  Logger.debug(`${TAG}/triggerShortcutSaga`, 'Initiating request to claim reward', payload)
+
+  const hooksApiUrl = yield select(hooksApiUrlSelector)
+
+  try {
+    const response = yield call(fetchWithTimeout, `${hooksApiUrl}/triggerShortcut`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    })
+    if (!response.ok) {
+      throw new Error(`Unable to trigger shortcut: ${response.status} ${response.statusText}`)
+    }
+
+    const { data } = yield call([response, 'json'])
+
+    const kit: ContractKit = yield call(getContractKit)
+    const walletAddress: string = yield call(getConnectedUnlockedAccount)
+    const normalizer = new TxParamsNormalizer(kit.connection)
+
+    Logger.debug(`${TAG}/triggerShortcutSaga`, 'Starting to claim reward(s)', data.transactions)
+
+    for (const transaction of data.transactions) {
+      applyChainIdWorkaround(transaction, yield call([kit.connection, 'chainId']))
+      const tx: CeloTx = yield call([normalizer, 'populate'], transaction)
+      const txo = buildTxo(kit, tx)
+
+      const receipt: CeloTxReceipt = yield call(
+        sendTransaction,
+        txo,
+        walletAddress,
+        newTransactionContext(TAG, 'Trigger shortcut')
+      )
+
+      Logger.debug(
+        `${TAG}/triggerShortcutSaga`,
+        'Claimed reward successful',
+        receipt.transactionHash
+      )
+    }
+
+    yield put(triggerShortcutSuccess(payload.id))
+    Toast.showWithGravity(
+      i18n.t('dappShortcuts.claimRewardsScreen.claimSuccess'),
+      Toast.SHORT,
+      Toast.BOTTOM
+    )
+    yield put(refreshAllBalances())
```

Comment:
No need to do it now, but I would be great if the code refreshing balances would listen to `triggerShortcutSuccess` too to decide to refresh all balances.
Small step to avoid the many dispatch anti-pattern we have in the code base.
https://redux.js.org/style-guide/#model-actions-as-events-not-setters

Reply:
yeah good shout, updated